mypy/checker.py

```diff
@@ -2506,6 +2506,68 @@ def conditional_type_map(expr: Expression,
         return {}, {}
 
 
+def is_callable_type(type: Optional[Type]) -> bool:
+    """Takes in a type and returns if that type is callable, that is,
+    if `type()` is well-typed (ignoring arguments)."""
+    callable_types = (FunctionLike, AnyType, TypeType)
+    if any(isinstance(type, callable_type) for callable_type in callable_types):
+        return True
+
+    if isinstance(type, UnionType):
+        return all(is_callable_type(item) for item in type.items)
+
+    if isinstance(type, TypeVarType):
+        return not is_callable_type(type.erase_to_union_or_bound())
+
+    if isinstance(type, Instance):
+        method = type.type.get_method('__call__')
+        if method:
+            return is_callable_type(method.type)
+
+    return False
+
+
+def conditional_callable_type_map(expr: Expression,
+                                  current_type: Optional[Type],
+                                  ) -> Tuple[TypeMap, TypeMap]:
+    """Takes in an expression and the current type of the expression
+
+    Returns a 2-tuple: The first element is a map from the expression to
+    the restricted type if it must be callable. The second element is a
```

Comment:
"must be" feels odd. Perhaps make the phrasing more similar to the docstring of conditional_type_map()?